src/vmq_queue.erl

```diff
@@ -296,6 +275,13 @@ drain(Event, State) ->
     lager:error("got unknown event in drain state ~p", [Event]),
     {next_state, drain, State}.
 
+drain({enqueue_many, Msgs}, _From, #state{drain_over_timer=TRef} =  State) ->
+    %% if multiple queues trigger queue migration at the 'same' time
+    %% so we've to queue those message otherwise they would be lost.
```

Comment:
you're right, the messages won't be lost, the other party will receive an error during the migration and transition to offline state.